libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -90,12 +121,18 @@ def _deserialize_links(json_blob: Optional[str]) -> Set[Link]:
 
 
 def _row_to_node(row: Any) -> Node:
-    metadata = _deserialize_metadata(row.metadata_blob)
+    metadata_s = row.metadata_s
+    if metadata_s is None:
+        metadata_s = {}
+    attributes_blob = row.attributes_blob
+    attributes_dict = (
+        _deserialize_metadata(attributes_blob) if attributes_blob is not None else {}
+    )
     links = _deserialize_links(row.links_blob)
     return Node(
         id=row.content_id,
         text=row.text_content,
-        metadata=metadata,
+        metadata={**attributes_dict, **metadata_s},
```

Comment:
I don't know that I'd combine the attributes and the metadata. It seems like for reconstructing the node we should *just* deserialize the attributes blob. When writing, we should put everything there (using that as the source of truth for round tripping), and then treat the `metadata_s` as just an index that supports queries.

Reply:
reverted to keeping all the metadata pickled in the `metadata_blob` and duplicating indexed fields in `metadata_s`